File: src/test/java/ch/grassl/day02/impl/MoveImporterTest.java
package ch.grassl.day02.impl;

import ch.grassl.common.Importer;
import ch.grassl.day02.impl.importer.MoveImporter;
import ch.grassl.day02.impl.model.Move;
import ch.grassl.day02.impl.model.Shape;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MoveImporterTest {

    protected static final String RESOURCE = "/day2_test.txt";

    private final Importer<Move> importer = MoveImporter.getInstance();

    @Test
    void testImportData() {
        List<Move> moves = importer.importData(RESOURCE);
        assertEquals(3, moves.size());
    }

    @Test
    void testMoves() {
        List<Move> moves = importer.importData(RESOURCE);
        assertEquals(new Move(Shape.ROCK, Shape.PAPER), moves.get(0));
        assertEquals(new Move(Shape.PAPER, Shape.ROCK), moves.get(1));
        assertEquals(new Move(Shape.SCISSORS, Shape.SCISSORS), moves.get(2));
    }

    @Test
    void testMovesScore() {
        List<Move> moves = importer.importData(RESOURCE);
        assertEquals(8, moves.get(0).getScore());
        assertEquals(1, moves.get(1).getScore());
        assertEquals(6, moves.get(2).getScore());
    }
}

File: src/test/java/ch/grassl/day/day09/impl/MoveImporterTest.kt
package ch.grassl.day.day09.impl

import ch.grassl.day.day09.impl.importer.MoveImporter
import ch.grassl.day.day09.impl.model.Direction
import ch.grassl.day.day09.impl.model.Move
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class MoveImporterTest {

    companion object {
        const val RESOURCE = "/day09/day09_test.txt"
    }

    private val importer = MoveImporter()

    @Test
    fun importData() {
        val moves = importer.importData(RESOURCE)
        assertEquals(Move(Direction.R, 4), moves[0])
        assertEquals(Move(Direction.U, 4), moves[1])
        assertEquals(Move(Direction.L, 3), moves[2])
        assertEquals(Move(Direction.D, 1), moves[3])
    }
}